src/finite_field.rs

```diff
@@ -28,7 +37,7 @@ impl std::ops::Add for Field {
     type Output = Field;
 
     fn add(self, rhs: Self) -> Self {
-        self - Field(MODULUS - rhs.0)
+        Self(SMALL_FP.add(self.0, rhs.0))
```

Comment:
This doesn't need to be `Self::from` because `Fp` already guarantees everything is done `mod p`, right?

Reply:
I'm not sure what your question is about. My understanding of what this line does: grab the `u128` underlying `self` and `rhs`, wrap the sum in a `Field`, and returns the result.

Reply:
Sorry, I was being unnecessarily terse. I was trying to confirm that this intentionally avoids invoking `Field::from()` to convert the result of `Fp:add()` into `Field`. In any case, I see now my question is a non sequitur, because there is no `Field::from<u128>()` that could be used here.